src/Orc.SystemInfo/Services/WindowsManagementInformationService.cs

```diff
@@ -23,42 +23,43 @@ public string GetIdentifier(string wmiClass, string wmiProperty, string addition
 
             var query = string.Format("SELECT {0}{1} FROM {2}", wmiProperty,
                 string.IsNullOrWhiteSpace(additionalWmiToCheck) ? string.Empty : string.Format(", {0}", additionalWmiToCheck), wmiClass);
-            var wmiSearcher = new ManagementObjectSearcher(query);
-            var managementObjectCollection = wmiSearcher.Get();
 
-            foreach (var managementObject in managementObjectCollection)
+            using (var connection = new WindowsManagementConnection())
```

Comment:
in the latest C# you can use:
`using var connection = new WindowsManagementConnection();`
it can simplify a code and make it more readable
because it will let you get rid of one level of {} incapsulation